File: haskell/insertionsort.hs
{-
	insertionsort
-}

insert n l = (filter (< n) l) ++ [n] ++ (filter (>= n) l)

insertionsort l = foldr insert [] l

main = 
    print (insertionsort [7,5,1,3,2,4,8,6,9,0])

File: prolog/insertionsort.pl
insert(X, [], [X]).
insert(X, [Head|Tail], [Head|Rest]):- X > Head, insert(X, Tail, Rest).
insert(X, [Head|Tail], [X, Head|Tail]):- X =< Head.

insertionsort([], []).
insertionsort(L, L).
insertionsort([X|L1], L2):- insert(X, L, L2), insertionsort(L1, L).